centroid_decompostition.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+class Graph 
+{
+private:
+	int ver;
+	vector<int>* adj;
```

Comment:
@KtlTheBest Could you clarify what your goal is here?

Currently, you are trying to set up a system in which you have an array of vectors. A better way to express that would be using:

```C++
int ver_; // Add underscore to end for members
std::vector<std::vector<int>> adj_;
std::vector<std::vector<int> centroidTree_;
std::vector<int> sizes_;
std::vector<bool> marked_; // Deleted is misleading, you may also want a different container, but it doesn't matter here
```

In your constructor using the member initialize syntax:

```C++
Graph(int num) : 
    ver_(num)
    adj_(num+ 1, std::vector<int>), // Don't rely on ver being initialized, even though it is
    centroidTree_(num+ 1, std::vector<int>),
    sizes_(num+ 1),
    marked_(num + 1) // Make sure you really mean num + 1
{}
```

Now you can ditch the constructor.